app/models/project.rb

```diff
@@ -12,10 +13,21 @@ class Project < ApplicationRecord
   has_many :tags, through: :taggings
   mount_uploader :image_preview, ImagePreviewUploader
 
+  include PgSearch
+  pg_search_scope :text_search, against: [:name, :description]
+
   self.per_page = 8
 
   acts_as_commontable
   # after_commit :send_mail, on: :create
+  def self.search(query)
+    if query.present?
+      Project.text_search(query)
+    else
+      Project.all
```

Comment:
Can we have default limit for this as well

Reply:
what do you think default limit should be set to ? According to me 50 is good enough.